packages/graphql-client/src/graphql-client/graphql-client.ts

```diff
@@ -165,3 +184,276 @@ function generateRequest(
+                          data && path ? buildDataObjectByPath(path, data) : {},
+                        ...getKeyValueIfValid("errors", errors),
+                      };
+                    });
+
+                  return {
+                    data:
+                      incrementalArray.length === 1
+                        ? incrementalArray[0].data
+                        : buildCombinedDataObject([
+                            ...incrementalArray.map(({ data }) => data),
+                          ]),
+                    ...getKeyValueIfValid(
+                      "errors",
+                      combineErrors(incrementalArray),
+                    ),
+                    hasNext,
+                  };
+                });
+
+              responseExtensions =
+                dataArray.find((datum) => datum.extensions)?.extensions ??
+                responseExtensions;
+
+              const responseErrors = combineErrors(dataArray);
+
+              combinedData = buildCombinedDataObject([
+                combinedData,
+                ...dataArray.map(({ data }) => data),
+              ]);
+
+              streamHasNext = dataArray.slice(-1)[0].hasNext;
+
+              if (responseErrors.length > 0) {
+                throw new Error(GQL_API_ERROR, {
+                  cause: {
+                    graphQLErrors: responseErrors,
+                  },
+                });
+              }
+
+              if (Object.keys(combinedData).length === 0) {
+                throw new Error(NO_DATA_OR_ERRORS_ERROR);
+              }
+
+              yield {
+                ...getKeyValueIfValid("data", combinedData),
+                ...getKeyValueIfValid("extensions", responseExtensions),
+                hasNext: streamHasNext,
+              };
+            }
+
+            if (streamHasNext) {
+              throw new Error(`Response stream terminated unexpectedly`);
+            }
+          } catch (error) {
+            const cause = getErrorCause(error);
+
+            yield {
+              ...getKeyValueIfValid("data", combinedData),
+              ...getKeyValueIfValid("extensions", responseExtensions),
+              errors: {
+                message: formatErrorMessage(getErrorMessage(error)),
+                networkStatusCode: status,
+                ...getKeyValueIfValid("graphQLErrors", cause?.graphQLErrors),
+                response,
+              },
+              hasNext: false,
+            };
+          }
+        },
+      };
+    } catch (error) {
+      return {
+        async *[Symbol.asyncIterator]() {
+          const response = getErrorCause(error);
+
+          yield {
+            errors: {
+              message: formatErrorMessage(getErrorMessage(error)),
+              ...getKeyValueIfValid("networkStatusCode", response?.status),
+              ...getKeyValueIfValid("response", response),
+            },
+            hasNext: false,
+          };
+        },
+      };
+    }
+  };
```

Comment:
The errors inside the async iterator will not be caught in the outer try/catch because the top level try/catch code has already executed. When the async iterator errors out, it needs to be handled internally.